### PR TITLE
ouch 0.3.1 (new formula)

### PR DESCRIPTION
**Pull request by bayandin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/ouch.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Ouch < Formula
+  desc "Painless compression and decompression for your terminal"
+  homepage "https://github.com/ouch-org/ouch"
+  url "https://github.com/ouch-org/ouch/archive/refs/tags/0.3.1.tar.gz"
+  sha256 "269abaf5ac2f80da3796dbf5e73419c1b64104d1295f3ff57965141f079e6f6d"
+  license "MIT"
+  head "https://github.com/ouch-org/ouch.git", branch: "main"
+
+  depends_on "rust" => :build
```
Comment (carlocab) [inline Formula/ouch.rb:9]:
```suggestion
  depends_on "rust" => :build
  uses_from_macos "bz2"
  uses_from_macos "xz"
  uses_from_macos "zlib"
```

File: Formula/ouch.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  uses_from_macos "bz2"
```
Comment (carlocab) [inline Formula/ouch.rb:11]:
```suggestion
  uses_from_macos "bzip2"
```
Oops

File: Formula/ouch.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  url "https://github.com/ouch-org/ouch/archive/refs/tags/0.3.1.tar.gz"
+  sha256 "269abaf5ac2f80da3796dbf5e73419c1b64104d1295f3ff57965141f079e6f6d"
+  license "MIT"
+  head "https://github.com/ouch-org/ouch.git", branch: "main"
+
+  depends_on "rust" => :build
+
+  uses_from_macos "bzip2"
+  uses_from_macos "xz"
+  uses_from_macos "zlib"
```
Comment (carlocab) [inline Formula/ouch.rb:13]: Ah, yea, these are needed but aren't related to the build error.

It looks like `cargo` is trying to link statically to a just-build Zstd (😞) and failing miserably.

**Comment by bayandin** [comment]
Ok, I don't really know what to do with a Linux build, I will close the PR, and maybe try again later

**Comment by carlocab** [comment]
I'm ok to just not bottle this on Linux first. No reason to deprive macOS users of a useful formula because we don't know how to build it on Linux.

If someone wants to install this formula on Linux, they can fix the build.

**Comment by bayandin** [comment]
Yep, that makes sense, let's do it!

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**